### PR TITLE
feat: Add environment support in config file and tag param

### PR DESCRIPTION
**Pull request by photowey**
Since the previous branch has been developed for too long and has many conflicts with the current master branch, a new PR is directly resubmitted

### CODE REVIEW

File: example/autowire_redis_client/cmd/main.go [hunk 1]
```diff
@@ -93,9 +102,28 @@ func (a *App) Run() {
...
+}
+
+func init() {
+	_ = os.Setenv("REDIS_ADDRESS_EXPAND", "localhost:6379")
 }
 
 func main() {
+	// wd, _ := os.Getwd()
+	// wd = filepath.Join(wd, "./autowire_redis_client/cmd")
+	// wd = filepath.Join(wd, "../conf")
```
Comment (LaurenceLiZhixin) [inline example/autowire_redis_client/cmd/main.go:126]: Can we remove these comments or describe them clearly.?

**Review by LaurenceLiZhixin — APPROVED** [review]

**Comment by LaurenceLiZhixin** [comment]
Thanks you for your pr, I think this design is necessary.

File: example/autowire_redis_client/cmd/main.go (hunk 1 above)
Comment (photowey) [reply example/autowire_redis_client/cmd/main.go]: I tested on both win and mac: `../conf` would point to a wrong path(e.g.: $GOPATH/src/github.com/alibaba/IOC-Golang/conf/ioc_golang.yaml). 

So there is a write of `wd, _ := os.Getwd()`.

**Comment by LaurenceLiZhixin** [comment]
fix https://github.com/alibaba/IOC-golang/issues/55

**Pull request merged**